jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/LCAAlgorithm.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import org.jgrapht.alg.util.Pair;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Algorithm to compute a <a href="https://en.wikipedia.org/wiki/Lowest_common_ancestor">lowest common ancestor</a>
+ * in a tree, forest or DAG.
+ *
+ * @param <V> vertex the graph vertex type
+ *
+ * @author Alexandru Valeanu
+ */
+public interface LCAAlgorithm<V> {
```

Comment:
For the main interface name, let's spell it out as LowestCommonAncestorAlgorithm.  Elsewhere (e.g. in the method names and in the implementation classes) we can keep the acronym.